MSAL/src/native_auth/network/responses/validator/token/validated_response/MSALNativeAuthTokenValidatedResponse.swift

```diff
@@ -31,98 +31,189 @@ enum MSALNativeAuthTokenValidatedResponse {
-    case unsupportedChallengeType(message: String?)
-    case strongAuthRequired(message: String?)
-    case invalidScope(message: String?)
-    case authorizationPending(message: String?)
-    case slowDown(message: String?)
+    case expiredToken(MSALNativeAuthTokenResponseError)
+    case expiredRefreshToken(MSALNativeAuthTokenResponseError)
+    case unauthorizedClient(MSALNativeAuthTokenResponseError)
+    case invalidRequest(MSALNativeAuthTokenResponseError)
+    case unexpectedError(MSALNativeAuthTokenResponseError?)
+    case userNotFound(MSALNativeAuthTokenResponseError)
+    case invalidPassword(MSALNativeAuthTokenResponseError)
+    case invalidOOBCode(MSALNativeAuthTokenResponseError)
+    case unsupportedChallengeType(MSALNativeAuthTokenResponseError)
+    case strongAuthRequired(MSALNativeAuthTokenResponseError)
+    case invalidScope(MSALNativeAuthTokenResponseError)
+    case authorizationPending(MSALNativeAuthTokenResponseError)
+    case slowDown(MSALNativeAuthTokenResponseError)
 
-    func convertToSignInPasswordStartError() -> SignInStartError {
+    // swiftlint:disable:next function_body_length
+    func convertToSignInPasswordStartError(context: MSIDRequestContext) -> SignInStartError {
         switch self {
-        case .expiredToken(let message),
-             .authorizationPending(let message),
-             .slowDown(let message),
-             .invalidRequest(let message),
-             .invalidOOBCode(let message),
-             .unauthorizedClient(let message),
-             .unsupportedChallengeType(let message),
-             .invalidScope(let message):
-            return SignInStartError(type: .generalError, message: message)
+        case .expiredToken(let apiError),
+             .authorizationPending(let apiError),
+             .slowDown(let apiError),
+             .invalidOOBCode(let apiError),
+             .unauthorizedClient(let apiError),
+             .unsupportedChallengeType(let apiError),
+             .invalidScope(let apiError),
+             .invalidRequest(let apiError):
+            return SignInStartError(
+                type: .generalError,
+                message: apiError.errorDescription,
+                correlationId: context.correlationId(),
+                errorCodes: apiError.errorCodes ?? [],
+                errorUri: apiError.errorURI
+            )
         case .generalError:
-            return SignInStartError(type: .generalError)
-        case .userNotFound(let message):
-            return SignInStartError(type: .userNotFound, message: message)
-        case .invalidPassword(let message):
-            return SignInStartError(type: .invalidCredentials, message: message)
-        case .strongAuthRequired(let message):
-            return SignInStartError(type: .browserRequired, message: message)
-        case .expiredRefreshToken(let message):
+            return SignInStartError(type: .generalError, correlationId: context.correlationId())
```

Comment:
Can we return the same amount of information also for this generale error?

Reply:
Yes, we can. But I've seen in some cases the `.generalError` is returned without having an ApiError available, so I've set it as optional.